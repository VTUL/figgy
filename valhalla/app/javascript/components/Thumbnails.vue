<template>
  <div v-model="thumbnails" tag="div" name="list-complete" class="img_gallery">
      <div @click.capture="select(thumbnail.id, $event)"
            v-bind:style="{'max-width': thumbPixelWidth + 'px' }"
            class="thumbnail"
            v-bind:class="{ hasChanged: hasChanged(thumbnail.id) }"
            v-for="thumbnail in thumbnails" :key="thumbnail.id">
        <img :src="thumbnail.url" class="thumb">
        <div v-bind:style="{'padding': captionPixelPadding + 'px' }" class="caption">
          {{thumbnail.label}}
        </div>
      </div>
  </div>
</template>

<script>
export default {
  name: 'thumbnails',
  data: function () {
    return {
      thumbPixelWidth: 200,
      captionPixelPadding: 9
    }
  },
  computed: {
    thumbnails: {
      get () {
        return this.$store.state.images
      }
    },
    changeList: {
      get () {
        return this.$store.state.changeList
      }
    },
    selected: {
      get () {
        return this.$store.state.selected
      }
    }
  },
  methods: {
    hasChanged: function (id) {
      if (this.changeList.indexOf(id) > -1) {
        return true
      } else {
        return false
      }
    },
    isSelected: function (thumbnail) {
      return false
      // if (this.selected.indexOf(thumbnail) > -1) {
      //   return true
      // } else {
      //   return false
      // }
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>

.thumbnail {
  display: inline-block;
  max-width: 200px;
  margin: 10px;
}

.img_gallery {
  overflow: auto;
  height: calc(100% - 40px);
  border-radius: 4px;
  margin-bottom: 40px;
  clear: both;
}

.selected {
  border: 2px solid #9ecaed;
  box-shadow: 0 0 10px #9ecaed;
}

.hasChanged {
  background-color: Tomato
}

.thumbnail .caption {
  pointer-events: none;
  overflow: hidden;
}

.thumb {
  pointer-events: none;
}

</style>
