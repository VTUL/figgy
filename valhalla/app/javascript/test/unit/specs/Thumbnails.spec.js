import Vue from 'vue'
import Vuex from 'vuex'
import { mount, shallow, createLocalVue } from 'vue-test-utils'
import Thumbnails from '@/components/Thumbnails'
import Fixtures from '@/test/fixtures/image-collection'
const localVue = createLocalVue()
localVue.use(Vuex)

describe('Thumbnails.vue', () => {
  let wrapper
  let options
  let actions
  let store

  beforeEach(() => {
    actions = {
      actionClick: jest.fn(),
      actionInput: jest.fn()
    }
    store = new Vuex.Store({
      state: {
        images: Fixtures.imageCollection,
        selected: Fixtures.selected,
        ogImages: Fixtures.imageCollection,
        changeList: Fixtures.changeList
      },
      actions
    })

    options = {
      computed: {
        thumbnails: {
          get () {
            return state.imageCollection
          }
        },
        changelist: {
          get () {
            return state.changeList
          }
        },
        selected: {
          get () {
            return state.selected
          }
        }
      }
    }

  })

  it('renders a $store.state value return from computed', () => {
    const wrapper = shallow(Thumbnails, { options, store, localVue })
    let expanded = wrapper.find('.thumbnail')
    const thumb = expanded.html().replace(/(<(pre|script|style|textarea)[^]+?<\/\2)|(^|>)\s+|\s+(?=<|$)/g, "$1$3")
    expect(thumb).toEqual('<div class="thumbnail" style="max-width: 200px;"><img src="http://example.com" class="thumb"><div class="caption" style="padding: 9px;">baz</div></div>')
  })

  it('has the expected html structure', () => {
    const wrapper = shallow(Thumbnails, { options, store, localVue })
    expect(wrapper.element).toMatchSnapshot()
  })

  // it('selectedImageUrl function has the expected value', () => {
  //   expect(wrapper.vm.selectedImageUrl).toEqual("http://localhost:3000/image-service/50b5e49b-ade7-4278-8265-4f72081f26a5/full/400,/0/default.jpg")
  // })

})
