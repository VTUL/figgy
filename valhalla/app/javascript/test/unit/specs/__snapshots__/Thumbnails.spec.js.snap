// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbnails.vue has the expected html structure 1`] = `
<div
  class="img_gallery"
  name="list-complete"
  tag="div"
>
  <div
    class="thumbnail"
    style="max-width: 200px;"
  >
    <img
      class="thumb"
      src="http://example.com"
    />
     
    <div
      class="caption"
      style="padding: 9px;"
    >
      
        baz
      
    </div>
  </div>
  <div
    class="thumbnail"
    style="max-width: 200px;"
  >
    <img
      class="thumb"
      src="http://example.com"
    />
     
    <div
      class="caption"
      style="padding: 9px;"
    >
      
        p. foo
      
    </div>
  </div>
</div>
`;
